platform/mellanox/iproute2/Makefile

```diff
@@ -0,0 +1,49 @@
+SHELL = /bin/bash
+.ONESHELL:
+.SHELLFLAGS += -e
+
+MAIN_TARGET = iproute2-mlnx_$(IPROUTE2_MLNX_VERSION_FULL)_$(CONFIGURED_ARCH).deb
+IPROUTE2_DEB = iproute2_$(IPROUTE2_MLNX_VERSION_FULL)_$(CONFIGURED_ARCH).deb
+IPROUTE2_TMP := $(shell mktemp -u -d -t ipr.XXXXXXXXXX)
+
+$(addprefix $(DEST)/, $(MAIN_TARGET)): $(DEST)/% :
+	# Remove any stale files
+	rm -rf iproute2-$(IPROUTE2_MLNX_VERSION)
+
+	wget -O iproute2_$(IPROUTE2_MLNX_VERSION).orig.tar.xz http://deb.debian.org/debian/pool/main/i/iproute2/iproute2_$(IPROUTE2_MLNX_VERSION).orig.tar.xz
+	wget -O iproute2_$(IPROUTE2_MLNX_VERSION_FULL).dsc http://deb.debian.org/debian/pool/main/i/iproute2/iproute2_$(IPROUTE2_MLNX_VERSION_FULL).dsc
+	wget -O iproute2_$(IPROUTE2_MLNX_VERSION_FULL).debian.tar.xz http://deb.debian.org/debian/pool/main/i/iproute2/iproute2_$(IPROUTE2_MLNX_VERSION_FULL).debian.tar.xz
+	dpkg-source -x iproute2_$(IPROUTE2_MLNX_VERSION_FULL).dsc
+
+	if [ ! -z "$(MLNX_SDK_SOURCE_BASE_URL)" ]; then
+		# Build with SDK compiled from sources
+		python scripts/get_sdk_patches.py -i $(IPROUTE2_MLNX_VERSION) -u $(MLNX_SDK_SOURCE_BASE_URL)
+	else
+		# Build with SDK debians taken from github
+		python scripts/get_sdk_patches.py -i $(IPROUTE2_MLNX_VERSION) -s $(MLNX_SDK_VERSION)
+	fi
+
+	pushd iproute2-$(IPROUTE2_MLNX_VERSION)
+
+	# Create git repository to apply patches with stg
+	git init
+	git add -f *
+	git commit -qm "initial commit"
+
+	# Apply Patches
+	stg init
+	stg import -s ../patch/series
+
+	# Build debian
+	dpkg-buildpackage -us -uc -b -j$(SONIC_CONFIG_MAKE_JOBS) --admindir $(SONIC_DPKG_ADMINDIR)
+
+	# Rename debian
+	mkdir -p $(IPROUTE2_TMP)
+	mkdir -p $(IPROUTE2_TMP)/DEBIAN
+	dpkg-deb --extract ../$(IPROUTE2_DEB) $(IPROUTE2_TMP)
+	dpkg-deb --control ../$(IPROUTE2_DEB) $(IPROUTE2_TMP)/DEBIAN
+	sed -i -r 's/Package: iproute2/Package: iproute2-mlnx/' $(IPROUTE2_TMP)/DEBIAN/control
```

Comment:
Can't we do it even before building the package?

Reply:
Yeah, why can't we have a patch for over any debian changes, like we do for sonic-linux-kernel https://github.com/sonic-net/sonic-linux-kernel/tree/master/patch/preconfig

Reply:
Also, why do we need a TMP directory, why can't we extract in the same directory?

Reply:
@dgsudharsan according to debian [documentation ](https://www.debian.org/doc/manuals/debian-faq/pkg-basics.en.html) the package name is being determined by field "Package" in control file. I have tried to change this field  before compilation but then compilation fails, I spent a lot of time trying to make it work but I couldn't. On the other hand, the method I took works cleanly, I measured the time spent there, it takes 0.863 seconds. 

@vivekrnv current directory contains the downloaded iproute2 source code with a lot of files and folders. If I extract to current working directory it works but the packaging time increases from 0.863 sec to 8.316 sec.